prereqs.py

```diff
@@ -213,6 +215,69 @@ def wasm_checks(install, installed_rust_targets):
         print("rustup target add wasm32-unknown-unknown")
 
 
+def download_and_extract(url_base, tar_file, out_dir):
+    os.makedirs(out_dir, exist_ok=True)
+    temp_file = tempfile.gettempdir() + os.sep + tar_file
+
+    # Note: Using curl and tar as subprocesses rather than Python libraries for features such as --strip-components
+    subprocess.run(["curl", "-L", "-o", temp_file, url_base + tar_file], check=True)
+    subprocess.run(
+        ["tar", "-xzf", temp_file, "--strip-components=1", "-C", out_dir], check=True
+    )
+
+    os.remove(temp_file)  # Clean up the tar file
+
+
+def install_wasm_bindgen():
+    ver_str = ".".join(str(v) for v in wasm_bindgen_ver)
+    # Maintain the below mappings as filenames are inconsistent, and we want x64 builds on Windows ARM64
+    wasm_bindgen_tar_map = {
+        "darwin": {
+            "arm64": f"wasm-bindgen-{ver_str}-aarch64-apple-darwin.tar.gz",
+            "x64": f"wasm-bindgen-{ver_str}-x86_64-apple-darwin.tar.gz",
+        },
+        "linux": {
+            "arm64": f"wasm-bindgen-{ver_str}-aarch64-unknown-linux-gnu.tar.gz",
+            "x64": f"wasm-bindgen-{ver_str}-x86_64-unknown-linux-musl.tar.gz",
+        },
+        "windows": {
+            "arm64": f"wasm-bindgen-{ver_str}-x86_64-pc-windows-msvc.tar.gz",
+            "x64": f"wasm-bindgen-{ver_str}-x86_64-pc-windows-msvc.tar.gz",
+        },
+    }
+    wasm_bindgen_base_url = (
+        f"https://github.com/rustwasm/wasm-bindgen/releases/download/{ver_str}/"
+    )
+    wasm_bindgen_filename = wasm_bindgen_tar_map[platform_sys][platform_arch]
+    out_dir = os.path.expanduser("~/wasm-bindgen")
+
+    download_and_extract(wasm_bindgen_base_url, wasm_bindgen_filename, out_dir)
+    # File of interest will be in "~/wasm-bindgen/wasm-bindgen"
+
+
+def install_binaryen():
+    binaryen_tar_map = {
+        "darwin": {
+            "arm64": f"binaryen-version_{binaryen_ver}-arm64-macos.tar.gz",
+            "x64": f"binaryen-version_{binaryen_ver}-x86_64-macos.tar.gz",
+        },
+        "linux": {
+            "arm64": f"binaryen-version_{binaryen_ver}-aarch64-linux.tar.gz",
+            "x64": f"binaryen-version_{binaryen_ver}-x86_64-linux.tar.gz",
+        },
+        "windows": {
+            "arm64": f"binaryen-version_{binaryen_ver}-x86_64-windows.tar.gz",
+            "x64": f"binaryen-version_{binaryen_ver}-x86_64-windows.tar.gz",
+        },
+    }
+    binaryen_base_url = f"https://github.com/WebAssembly/binaryen/releases/download/version_{binaryen_ver}/"
```

Comment:
I did mess around with using the cross-platform 'binaryen' npm package, but running the wasm optimization was taking about 47sec on my mac, versus about 3sec if using the platform native binaries - so installing the binaries it is!